sn_networking/src/record_store.rs

```diff
@@ -198,7 +195,36 @@ impl DiskBackedRecordStore {
         Ok(())
     }
 
-    fn storage_pruning(&mut self) {
+    /// Prune the records in the store to ensure that we free up space
+    /// for new records.
+    ///
+    /// This can happen if we're full, or once in every X PUTs
+    ///
+    /// An error is returned if we prune and we're _still_ at max_record count
+    fn prune_storage_if_needed(&mut self) -> Result<()> {
+        let num_records = self.records.len();
+        let mut should_prune = num_records >= self.config.max_records;
+        if !should_prune {
+            // Use os rng to call ensure we prune at least PRUNE_ONCE_EVERY_PUTS
```

Comment:
By generating a number in the `0..100` range, it's not 'ensured' that `should_prune` is `true` once in max. 100 times. Statistically it could very well happen `0` isn't generated for a while.

Reply:
Indeed, it could also happen one after another. It will average out though.

We could just modulo here though perhaps and get a decent check interval actually

Reply:
Aye that's cleaner